src/Core/Checkout/Promotion/Subscriber/PromotionIndividualCodeRedeemer.php

```diff
@@ -104,4 +126,32 @@ private function getIndividualCode(string $code, Context $context): PromotionInd
 
         return $promotion;
     }
+
+    private function collectLineItems(EntityWrittenEvent $event): OrderLineItemCollection
+    {
+        $orderLineItems = new OrderLineItemCollection();
+
+        foreach ($event->getWriteResults() as $result) {
+            if ($result->getPayload()['type'] !== 'promotion') {
```

Comment:
in line 69 (nice) you are also checking for the line item type, which seems duplicate. Also, please check against the constant instead of a new string.